### DISCUSSION TITLE
Allow for use of TrueType Fonts

### DISCUSSION THREAD

**Discussion by JoshuaPK** [Motion-Project/motion#1497] [General]
I often like to adjust the size for the text overlays so that I can read the text while I am viewing a low-resolution stream of my camera.  When I then switch to full-screen full-resolution viewing, the text looks jagged and not clean.  It would be helpful if we could use a clean scaleable font for the text overlay, such as the freely available IBM Nouveau TrueType font.

**Comment by tosiara** [comment]
Could you provide sample pictures please?

**Comment by angrycoding** [comment]
> I often like to adjust the size for the text overlays so that I can read the text while I am viewing a low-resolution stream of my camera. When I then switch to full-screen full-resolution viewing, the text looks jagged and not clean. It would be helpful if we could use a clean scaleable font for the text overlay, such as the freely available IBM Nouveau TrueType font.

How can TrueType font can solve your problem? :)
Do you think that text rendered with true type font will scale properly along with the video?

**Comment by JoshuaPK** [comment]
@angrycoding - if I convert the TrueType glyph into a 9x16 pixel bitmap, then scale up the bitmap- no, it won't scale properly.  But if I want larger text, I would not manually scale the 9x16 bitmap up... I would convert the TT glyph to a 18x32 or 36x64 bitmap.  In the attached photo, note how the text overlay from the camera itself looks compared to the text overlay provided by Motion.
![MotionTest-Large](https://user-images.githubusercontent.com/4734265/71384904-285fe100-25b2-11ea-9337-893bcc368ae4.jpg)

**Comment by angrycoding** [comment]
@JoshuaPK I'm still not getting it. Text in the upper left corner of your picture is overlayed by the camera itself, during encoding. Software that is responsible for overlaying it on the camera level during encoding - knows what is the dimensions of the stream, so it can adjust font size. Still don't understand how TrueType font would solve this:

> When I then switch to full-screen full-resolution viewing, the text looks jagged and not clean

**Comment by JoshuaPK** [comment]
Here's where this all began.  I'm running a MotionEye server to view cameras at a remote location.  The main MotionEye page provides a view of all of the cameras at once.  In the communication between the cameras (RTSP network cameras) and the Motion daemon, the cameras provide a feed at 1920x1080.  Something then scales those images down to approx 640x360.  In order to clearly see the text in those 640x360 views, I have to configure the text size in MotionEye (which then configures the text size in the Motion daemon) to be large.  If I am reading and understanding the code correctly, the Motion daemon uses the bitmaps in draw.c and scales the bitmaps up from 7x8 to whatever is needed.  Instead of taking static bitmaps and scaling them up (resulting in the jagged edges), my thought is that the Motion daemon could simply generate the text in the correct size to begin with, using the routines in the FreeType library.  The "Nouveau IBM" font is a good font for this purpose.

**Comment by JoshuaPK** [comment]
I am not proposing to require the use of the FreeType library.  I am proposing conditional compilation so that current users can compile as normal without change in the code, and folks who want to use TrueType can set that compile-time option to compile in the FreeType support.
